### PR TITLE
cloudpan189-go: init at 0.1.3

### PR DESCRIPTION
**Pull request by xddxdd**
## Description of changes

CLI for China Telecom 189 Cloud Drive service, implemented in Go.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/cl/cloudpan189-go/package.nix [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  dontUnpack = true;
+  postInstall = ''
+    install -Dm755 ${startScript} $out/bin/cloudpan189-go
+  '';
+
+  meta = with lib; {
+    maintainers = with lib.maintainers; [ xddxdd ];
+    description = "CLI for China Telecom 189 Cloud Drive service, implemented in Go";
+    homepage = "https://github.com/tickstep/cloudpan189-go";
+    license = licenses.asl20;
```
Comment (FliegendeWurst) [inline pkgs/by-name/cl/cloudpan189-go/package.nix:51]:
```suggestion
  meta = {
    maintainers = with lib.maintainers; [ xddxdd ];
    description = "CLI for China Telecom 189 Cloud Drive service, implemented in Go";
    homepage = "https://github.com/tickstep/cloudpan189-go";
    license = lib.licenses.asl20;
```

File: pkgs/by-name/cl/cloudpan189-go/package.nix [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+let
+  pname = "cloudpan189-go";
+  version = "0.1.3";
+  cmd = buildGoModule {
+    inherit pname version;
+    src = fetchFromGitHub {
+      owner = "tickstep";
+      repo = "cloudpan189-go";
+      rev = "v${version}";
+      fetchSubmodules = false;
```
Comment (FliegendeWurst) [inline pkgs/by-name/cl/cloudpan189-go/package.nix:18]:
```suggestion
```
This is the default.

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]
Build fails
```
Running phase: buildPhase
Building subPackage .
# github.com/tickstep/cloudpan189-go
link: github.com/tickstep/library-go/cachepool: invalid reference to runtime.rawbyteslice
```

**Comment by xddxdd** [comment]
Rebased on latest master and fixed building by using Go 1.22.

File: pkgs/by-name/cl/cloudpan189-go/package.nix [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    vendorHash = "sha256-6t4wJqUGJneR6Hv7Dotr4P9MTA1oQcCe/ujDojS0g8s=";
+
+    # Dirty way to fix dependency issue
+    overrideModAttrs = _: {
+      postInstall = ''
+        sed -i '/go:linkname/d' $out/github.com/tickstep/library-go/expires/expires.go
+      '';
+    };
+
+    doCheck = false;
```
Comment (FliegendeWurst) [inline pkgs/by-name/cl/cloudpan189-go/package.nix:30]:
```suggestion
    doCheck = true;
```

File: pkgs/by-name/cl/cloudpan189-go/package.nix [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  lib,
+  fetchFromGitHub,
+  stdenvNoCC,
+  writeShellScript,
+  versionCheckHook,
+}:
+let
+  pname = "cloudpan189-go";
+  version = "0.1.3";
+  cmd = buildGo122Module {
```
Comment (FliegendeWurst) [inline pkgs/by-name/cl/cloudpan189-go/package.nix:12]: Why is this outsourced? You could have this as the main package expression.

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/cl/cloudpan189-go/package.nix [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  overrideModAttrs = _: {
+    postInstall = ''
+      sed -i '/go:linkname/d' $out/github.com/tickstep/library-go/expires/expires.go
+    '';
+  };
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+  doInstallCheck = true;
```
Comment (FliegendeWurst) [inline pkgs/by-name/cl/cloudpan189-go/package.nix:28]:
```suggestion
```
Has no effect currently.

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**